### PR TITLE
feat: show commit hash of current build in flags menu

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

this adds two new features to the feature flags page

* the git short commit hash of the current build
* the latest production tag of the repo

this is intended for @shapeshift/operations to use to verify that

* they are testing the most recent version of the `release` branch
* that a release has successfully deployed to production

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

n/a - quality of life

## Risk

* the build fails to succeed (engineering won't merge this PR)
* this shows incorrect/misleading information

### Engineering

* this will not work locally with the dev server - the static files aren't available nor relevant via webpack
* you can test a locally served build via `yarn build && serve -s build -l 4000` if you comment out the `isLocalhost` check
* add `./metadata.json` to the ephemeral environment url path to review the served file, or see it from the network tab
* **the build command in fleek has been updated** to include a `git fetch --tags` step so it's now `apt update && apt --yes install curl wget git && git fetch --tags && npm i -g n && n 16.14.2 && yarn install --frozen-lockfile && yarn env develop && yarn build`
  * **please verify this correct, with the correct `yarn env <environment>` across all environments before merging**

### Operations

* this is mainly intended to be used on the [release](https://github.com/shapeshift/web/commits/release) and [main](https://github.com/shapeshift/web/commits/main) (production) branches
* this will **not** show the version number of the release you are currently testing - this will show the *latest production release*
* when the `Latest tag` shown on the flags page matches [the latest version number shown here](https://github.com/shapeshift/web/tags), you can be sure the release has successfully gone to production
* when you are testing this PR as part of the release, the `Commit hash` shown in the flags menu should match the commit hash [top of the list of commits on the release branch here](https://github.com/shapeshift/web/commits/release)

## Screenshots (if applicable)

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "show-linked-packages": "find node_modules node_modules/@* -maxdepth 1 -type l -print",
     "link-packages": "./scripts/link-packages.sh",
     "unlink-packages": "find node_modules node_modules/@* -maxdepth 1 -type l -print | awk -F/ '{ printf \"%s/%s\\n\",$2,$3; }' | xargs yarn unlink && yarn install --force",
-    "build": "NODE_OPTIONS=--max-old-space-size=4096 IMAGE_INLINE_SIZE_LIMIT=0 INLINE_RUNTIME_CHUNK=false DISABLE_ESLINT_PLUGIN=true react-app-rewired build && node ./scripts/writeHeaders.js && ./scripts/sha256sums.sh && ./scripts/verifyWebpackHashes.sh",
+    "build": "NODE_OPTIONS=--max-old-space-size=4096 IMAGE_INLINE_SIZE_LIMIT=0 INLINE_RUNTIME_CHUNK=false DISABLE_ESLINT_PLUGIN=true react-app-rewired build && node ./scripts/writeHeaders.js && ./scripts/sha256sums.sh && ./scripts/verifyWebpackHashes.sh && ts-node --skipProject ./scripts/writeBuildMetadata.ts",
```
Comment (0xdef1cafe) [inline package.json:9]: spit this out after we know the build succeeded

File: src/pages/Flags/Debugging.tsx [hunk 2]
```diff
@@ -1,28 +1,85 @@
...
+    latestTag: string
+  }
+
+  const [buildMetadata, setBuildMetadata] = useState<BuildMetadata | undefined>()
+  const isLocalhost = window.location.hostname === 'localhost'
+
+  useEffect(() => {
+    if (isLocalhost) return
+    ;(async () => {
+      const url = './metadata.json'
```
Comment (0xdef1cafe) [inline src/pages/Flags/Debugging.tsx:54]: grab the bits we spat out from the build

File: src/pages/Flags/Debugging.tsx [hunk 3]
```diff
@@ -37,13 +41,46 @@ export const Debugging = () => {
     }
   }, [])
 
+  type BuildMetadata = {
+    headShortCommitHash: string
+    latestTag: string
+  }
+
+  const [buildMetadata, setBuildMetadata] = useState<BuildMetadata | undefined>()
+  const isLocalhost = window.location.hostname === 'localhost'
```
Comment (gomesalexandre) [inline src/pages/Flags/Debugging.tsx:50]: Probably time to export this to its own util

**Review by gomesalexandre — APPROVED** [review]
- Tested with served built dist, with `isLocalhost` 🐒 'd to false

image
image
image
image
image

- Confirmed `git fetch --tags` is present as part of the build command in all Fleek envs
- Tested built dist with `dev`, `develop`, `private` and `app` environments, all looking good locally with the same app output as the screenshot above
- Ephemeral fleek env for this PR is also matching the screenshot above

**Review by gomesalexandre — APPROVED** [review]
Tested the ephemeral Fleek build again after updating branch, commit hash is matching the new one, and latest tag is still matching the latest release tag.

**Pull request merged**